src/Helper/BasePath.php

```diff
@@ -4,48 +4,61 @@
 
 namespace Laminas\View\Helper;
 
-use Laminas\View\Exception;
+use Laminas\View\Exception\RuntimeException;
 
 use function ltrim;
 use function rtrim;
+use function sprintf;
 
 /**
  * Helper for retrieving the base path.
+ *
+ * @final
  */
 class BasePath extends AbstractHelper
 {
-    /**
-     * Base path
-     *
-     * @var string
-     */
+    use DeprecatedAbstractHelperHierarchyTrait;
+
+    /** @var string|null */
     protected $basePath;
 
+    public function __construct(?string $basePath = null)
+    {
+        $this->basePath = $basePath;
+    }
+
     /**
      * Returns site's base path, or file with base path prepended.
      *
      * $file is appended to the base path for simplicity.
      *
      * @param  string|null $file
-     * @throws Exception\RuntimeException
+     * @throws RuntimeException
      * @return string
      */
     public function __invoke($file = null)
     {
-        if (null === $this->basePath) {
-            throw new Exception\RuntimeException('No base path provided');
+        if ($this->basePath === null) {
+            throw new RuntimeException('No base path provided');
         }
 
-        if (null !== $file) {
-            $file = '/' . ltrim($file, '/');
+        if (! empty($file)) {
```

Comment:
`empty()` should generally never be used 😬

Reply:
😂 I thought you'd hate that! I don't really see the problem with it. I know it's a string, or it's null, and I just want to check for a non-empty-string - I reckon it's more readable than the alternative `if ($file !== null && $file !== '') {`… `(string) $file !== ''` doesn't work here for psalm.